src/index.js

```diff
@@ -97,78 +109,95 @@ async function bootstrap(state) {
 
   if (state.seq > 0 && state.bootstrapDone === true) {
     await setSettings(mainIndex);
-    log.info('⛷ Bootstrap: done');
+    log.info('☑️   Bootstrap: done');
     return state;
   }
 
-  if (state.bootstrapLastId) {
-    log.info('⛷ Bootstrap: starting at doc %s', state.bootstrapLastId);
-    await loadHits();
-    return loop(state.bootstrapLastId);
-  } else {
-    const { taskID } = await client.deleteIndex(c.bootstrapIndexName);
-    await bootstrapIndex.waitTask(taskID);
-    log.info('⛷ Bootstrap: starting from the first doc');
-    const { seq } = await npm.info();
+  await loadHits();
+
+  const { seq, nbDocs: totalDocs } = await npm.info();
+  if (!state.bootstrapLastId) {
+    // Start from 0
+    log.info('⛷   Bootstrap: starting from the first doc');
     // first time this launches, we need to remember the last seq our bootstrap can trust
     await stateManager.save({ seq });
     await setSettings(bootstrapIndex);
-    await loadHits();
-    return loop(state.bootstrapLastId);
+  } else {
+    log.info('⛷   Bootstrap: starting at doc %s', state.bootstrapLastId);
   }
 
-  function loop(lastId) {
-    const options =
-      lastId === undefined
-        ? {}
-        : {
-            startkey: lastId,
-            skip: 1,
-          };
-
-    return db
-      .allDocs({
-        ...defaultOptions,
-        ...options,
-        limit: c.bootstrapConcurrency,
-      })
-      .then(async res => {
-        if (res.rows.length === 0) {
-          log.info('⛷ Bootstrap: done');
-          await stateManager.save({
-            bootstrapDone: true,
-            bootstrapLastDone: Date.now(),
-          });
+  log.info('-----');
+  log.info(`Total packages   ${totalDocs}`);
+  log.info('-----');
 
-          return moveToProduction();
-        }
+  let still = true;
+  while (still) {
+    still = await loop(state.bootstrapLastId);
```

Comment:
```suggestion
    lastProcessedId = await loop(state.bootstrapLastId);
```

A little confusing to mix the return type (boolean) with string, maybe something better can be found?

Reply:
indeed, I will use `state.bootstrapLastId` for clarity 👍